nilearn/input_data/base_masker.py

```diff
@@ -97,6 +97,11 @@ def filter_and_extract(imgs, extraction_function,
                                 func_memory_level=2,
                                 memory_level=memory_level)(imgs)
 
+    if np.any(np.isnan(region_signals)):
+        ensure_finite = True
+    else:
+        ensure_finite = False
```

Comment:
I think that we should always have ensure_finite=True here: it costs as much to check if there is a problem as to fix it.

I'll push these changes directly in your branch, to save time.